protobufs/types.proto

```
@@ -8,124 +8,142 @@ package proto;
-message ProtoControllerTrigger {
-  string type = 1; // Fixed value "ControllerTriggers"
-  int32 slot = 2; // Slot number (0-3)
-  int32 trigger = 3; // Trigger number (0 for left, 1 for right)
-  int32 value = 4; // trigger value (-32768 to 32767)
+// ControllerStateBatch - single message containing full or partial controller state
+message ProtoControllerStateBatch {
+  int32 session_slot = 1; // Session specific slot number (0-3)
+  string session_id = 2; // Session ID of the client
+
+  enum UpdateType {
+    FULL_STATE = 0;    // Complete controller state
```

Buf (1.59.0)

Enum value name "FULL_STATE" should be prefixed with "UPDATE_TYPE_".

(ENUM_VALUE_PREFIX)